### ISSUE TITLE
Multiple spaces in messages not shown as such

### ISSUE THREAD

**Issue by PolarizedIons** [thelounge/lounge#1229]
When there are multiple spaces in a message (like in motd ASCII art), lounge only shows it as one, messing up the alignment.

Example: lounge on the left and hexchat on the right, showing EsperNet's motd
![](https://i.imgur.com/jOupDen.png)

**Comment by DanielOaks** [comment]
This also affects the MAP command particularly – since multiple spaces don't get displayed as such, it pretty much becomes unreadable :P

**Comment by YaManicKill** [comment]
Well, it's html, so no matter how many spaces you have next to it, the browser will show it as 1 space.

We'll want to replace them with non-breaking spaces.

**Comment by starquake** [comment]
I think that breaks word-wrapping. May I suggest another fix? I think the correct way would be to use:
```
#chat .text {
  white-space: pre-wrap;
}
```
From MDN
[Sequences of whitespace are preserved. Lines are broken at newline characters, at <br>, and as necessary to fill line boxes.](https://developer.mozilla.org/en/docs/Web/CSS/white-space?v=control)

But that means all the texts going in that SPAN can not have newlines that aren't supposed to be there.

**Comment by mestaritonttu** [comment]
I tested with [KiwiIRC's next client](https://kiwiirc.com/nextclient/) and it has:

```
.kiwi-messagelist-body {
    display: block;
    white-space: pre-wrap;
    word-wrap: break-word;
        overflow-wrap: break-word;
    margin-left: 130px;
}
```

**Comment by astorije** [comment]
I could swear we've had this issue before, and fixed it! Is there someone courageous to look through The Lounge / Shout commit / PR histories? :D

> I tested with KiwiIRC's next client

(Interesting how in some ways it _really_ looks like The Lounge visually :D)

**Comment by YaManicKill** [comment]
> I could swear we've had this issue before, and fixed it! 

Yes we did, and it just came full circle. It was fixed in #239 

And I tried adding it back on locally and found ... that the bug we had with the table layout was caused by this (where it would wrap unnecessarily). It was then removed in #523 (specifically [this commit](https://github.com/thelounge/lounge/pull/523/commits/fb672ab57ff2a6d5ae8a54b2acef64d4e4354513)) so that makes sense.

**Comment by starquake** [comment]
@YaManicKill That happens because with pre-wrap, newlines get meaning and there are newlines in that part. And the newlines are there because there are newlines in the .tpl

I'm about to experiment with it and see if I can get a working situation. One that looks okay and also allows the expected copy+paste behaviour.

**Comment by YaManicKill** [comment]
> @YaManicKill That happens because with pre-wrap, newlines get meaning and there are newlines in that part. And the newlines are there because there are newlines in the .tpl

Yeah, I literally just came to that realisation... that took me far too long to figure out.

Alrighty, well, I guess if we want to do that, we'll need to strip out the newlines from the tpls but that doesn't feel nice.

It would be nice if we could get a solution without having to do this.

**Comment by starquake** [comment]
How about letting the template parser remove all newlines in the template?

**Comment by YaManicKill** [comment]
Aha, handlebars handles this.

From [here](http://handlebarsjs.com/expressions.html):

> Template whitespace may be omitted from either side of any mustache statement by adding a ~ character by the braces. When applied all whitespace on that side will be removed up to the first handlebars expression or non-whitespace character on that side.
> ```
> {{#each nav ~}}
>   <a href="{{url}}">
>     {{~#if test}}
>       {{~title}}
>     {{~^~}}
>       Empty
>     {{~/if~}}
>   </a>
> {{~/each}}
> ```

> with this context:
> 
> ```
> {
>   nav: [
>     {url: 'foo', test: true, title: 'bar'},
>     {url: 'bar'}
>  ] 
> }
> ```
> results in output sans newlines and formatting whitespace:
> `<a href="foo">bar</a><a href="bar">Empty</a>`

So we should just be able to add the '~' in the tpl and it'll remove the newlines when parsing. We will need to check that doesn't remove whitespace from the actual message (it says 'template whitespace' so that implies it won't to me).

**Comment by starquake** [comment]
Added
```
#chat .msg {
	user-select: all;
}
```
Editing tpl's and checking by hand... Not sure if everything is correct. Can't test kickban for example :-)

**Comment by starquake** [comment]
Remove `user-select:all`, as it has nothing to do with this. 

Here's a pull request #1237. Was unable to test some of the templates. And there are still some issues:
`(dd23e3b1@gateway/web/thelounge/ip.221.35.227.177) has quit(Client Quit)`

**Comment by astorije** [comment]
Fixed by #1242
